lib/rubycritic/cli/application.rb

```diff
@@ -14,12 +14,29 @@ def initialize(argv)
 
       def execute
         parsed_options = @options.parse
-        ::Rubycritic::CommandFactory.create(parsed_options).execute
+        result = ::Rubycritic::CommandFactory.create(parsed_options).execute
+
+        return STATUS_ERROR unless satisfy_minimum_score?(result, parsed_options)
+
         STATUS_SUCCESS
       rescue OptionParser::InvalidOption => error
         $stderr.puts "Error: #{error}"
         STATUS_ERROR
       end
+
+      private
+
+      def satisfy_minimum_score?(result, options)
+        return true unless result.is_a?(Hash) && result[:score]
```

Comment:
Addendum: In reference to my comment above, this whole logic should not be here but in a Reporter.
Without any type check of course ;)